app/src/main/java/space/taran/arknavigator/ui/fragments/SettingsFragment.kt

```diff
@@ -0,0 +1,196 @@
+                    negButtonText = R.string.no,
+                    posButtonCallback = { presenter.onResetPreferencesClick() }
+                )
+            }
+        }
+    }
+
+    override fun init() {
+        Log.d(SETTINGS_SCREEN, "initializing SettingsFragment")
+        (activity as MainActivity).setSelectedTab(2)
+        (activity as MainActivity).setToolbarVisibility(false)
+        (requireActivity() as MainActivity).setBottomNavigationVisibility(true)
+    }
+
+    override fun setCrashReportPreference(crashReport: CrashReport) {
+        binding.crashGroup.check(buttonFromCrashReport(crashReport))
+    }
+
+    override fun setImgCacheReplicationPref(imgReplicationPref: ImgCacheReplication) {
+        binding.imgReplicationGroup.check(buttonFromImgReplication(imgReplicationPref))
+    }
+
+    override fun setIndexReplicationPref(indexReplication: IndexReplication) {
+        binding.indexReplicationGroup.check(buttonFromIndexReplication(indexReplication))
+    }
+
+    override fun notifyUser(message: String, moreTime: Boolean) {
+        Notifications.notifyUser(context, message, moreTime)
+    }
+
+    private fun rgButtonSelected(rg: RadioGroup, checkedID: Int) {
+        rg.children.filterIsInstance(RadioButton::class.java).forEach { rb ->
+            if (rb.id != checkedID) rb.setTextColor(getColor(R.color.colorPrimary))
+            else rb.setTextColor(getColor(R.color.white))
+        }
+    }
+
+    private fun getColor(@ColorRes colorID: Int) =
+        ContextCompat.getColor(requireContext(), colorID)
+
+    private fun crashReportFromButton(buttonID: Int): CrashReport {
+        val reversed = crashReportMap.entries.associate { (k, v) -> v to k }
+        return reversed.getOrDefault(buttonID, CrashReport.NONE)
+    }
+
+    private fun buttonFromCrashReport(crashReport: CrashReport): Int {
+        return crashReportMap[crashReport]
+            ?: throw AssertionError("CrashReport must be of known type")
+    }
+
+    private fun imgReplicationFromButton(buttonID: Int): ImgCacheReplication {
+        val reversed = imgReplicationMap.entries.associate { (k, v) -> v to k }
+        return reversed.getOrDefault(buttonID, ImgCacheReplication.NONE)
+    }
+
+    private fun buttonFromImgReplication(imgReplication: ImgCacheReplication): Int {
+        return imgReplicationMap[imgReplication]
+            ?: throw AssertionError("ImgCacheReplication must be of known type")
+    }
+
+    private fun indexReplicationFromButton(buttonID: Int): IndexReplication {
+        val reversed = indexReplicationMap.entries.associate { (k, v) -> v to k }
+        return reversed.getOrDefault(buttonID, IndexReplication.NONE)
+    }
+
+    private fun buttonFromIndexReplication(indexReplication: IndexReplication): Int {
+        return indexReplicationMap[indexReplication]
+            ?: throw AssertionError("IndexReplication must be of known type")
+    }
+
+    private val crashReportMap = mapOf(
```

Comment:
From the names of these maps, it's not clear what exactly we map.
Good name should involve semantics of keys or values, or both.
Here we only know to what component the map relates.